server/util.ts

```diff
@@ -1,18 +1,17 @@
 import osu from "node-osu";
 
 import { IUser } from "./models/user";
+import { UserAuth } from "./permissions/UserAuth";
+import { UserRole } from "./permissions/UserRole";
 
 // Shared utilities used across the backend
 
 export const getOsuApi = () => new osu.Api(process.env.OSU_API_KEY, { parseNumeric: true });
 
-export const checkPermissions = (user: IUser, tourney: string, roles: string[]) => {
-  return (
-    user &&
-    user.username &&
-    (user.admin ||
-      user.roles.some(
-        (r) => ["Host", "Developer", ...roles].includes(r.role) && r.tourney == tourney
-      ))
-  );
+/**
+ * @deprecated Use UserAuth directly instead
+ */
+export const checkPermissions = async (user: IUser, tourney: string, roles: string[]) => {
```

Comment:
missed one `async` to remove

Reply:
Removed in 5d4d5ca3b2b9bdae4b43020c9893782bc5603af2